virttest/vt_cluster/__init__.py

```
@@ -0,0 +1,301 @@
+            pickle.dump(self._data, f, protocol=0)
+
+    def _load(self):
+        """
+        Load cluster state from the persistent storage file.
+
+        :return: The de-serialized cluster data dictionary containing
+                 partitions and nodes.
+        :rtype: dict
+        """
+        with open(self._filename, "rb") as f:
+            return pickle.load(f)
```

Ruff (0.12.2)

`pickle` and modules that wrap it can be unsafe when used to deserialize untrusted data, possible security issue

(S301)